Project_2/djangoProject/cars/views.py

```diff
@@ -23,15 +29,19 @@ def get_queryset(self):
         return Car.objects.all()
 
 
-# def detail(request, car_id):
-#     """
-#     Return a list with all the valid attributes of a car.
-#     """
-#     car = get_object_or_404(Car, pk=car_id)
-#     for _ in inspect.getmembers(car):
-#         if not _[0].startswith('_'):
-#             if not inspect.ismethod(_[1]):
-#                 return _
+def more_details(request, pk):
+    car = get_object_or_404(Car, pk=pk)
+    try:
+        car_detail = Car.objects.filter(pk=pk).values(*detail_list)[0]
```

Comment:
With this and the code at line 33, you are doing the same query twice to the DB, which is not efficient. I would recommend something like this to have fewer queries:
`
car_queryset = Car.objects.filter(pk=pk)
car = car_queryset.first()
try:
    car_detail = car_queryset.values(*detail_list)[0]
...
`